packages/salesforce-adapter/src/filters/utils.ts

```diff
@@ -375,23 +375,37 @@ export const extractFlatCustomObjectFields = async (elem: Element): Promise<Elem
     : [elem]
 )
 
-export const getWhereConditions = (
-  conditionSets: Record<string, string>[],
-  maxLen: number
+type QueryOperator = '>' | '<'
+type ComplexQuery = [string, QueryOperator, string]
+
+const getWhereConditions = (
+  exactConditionSets: Record<string, string>[],
+  complexConditionSets: ReadonlyArray<ComplexQuery>,
+  maxLen: number,
 ): string[] => {
-  const keys = _.uniq(conditionSets.flatMap(Object.keys))
+  const keys = _.uniq(exactConditionSets.flatMap(Object.keys))
   const constConditionPartLen = (
     _.sumBy(keys, key => `${key} IN ()`.length)
     + (' AND '.length * (keys.length - 1))
   )
 
+  const complexConditionQuery = complexConditionSets
+    .map(([field, op, value]) => `${field} ${op} ${value}`)
+    .join(' AND ')
+  const complexConditionsLength = complexConditionQuery.length
+    + ((Object.keys(exactConditionSets).length > 0 && complexConditionQuery.length > 0) ? ' AND '.length : 0)
+
+  if (complexConditionsLength > maxLen) {
+    throw new Error('Complex conditions need to be included in every query, thus can`t be longer than the maximum query length')
```

Comment:
```suggestion
    log.error('the WHERE part of the built query exceeds the max query length. getWhereConditions params: %o', { exactConditionSets, complexConditionSets, maxLen}')
    throw new Error('Built SOQL query too long')
```
If we reached this flow, I agree this should be an unrecoverable error. However, we need to make sure that the visible error to the users does not contain implementation details and needs to be understandable.
On our end, we need info on how we reached this flow as this is most likely coding error, as it's not affected by user input (at-least for now)